### PR TITLE
Corrected the reading of the port. See: #834

### LINKED ISSUE #834: [Bug] Issue with Overriding Default Server Port (8080) in application.properties"
The default port (8080) cannot be changed in `application.properties`.
`server.port=8443`

Pipeline and API interface cannot be used.

My Java knowledge is not very good, so here is my suggestion:

https://github.com/Stirling-Tools/Stirling-PDF/blob/main/src/main/java/stirling/software/SPDF/SPdfApplication.java

```java
package stirling.software.SPDF;

import java.nio.file.Files;
import java.nio.file.Paths;
import java.util.Collections;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.core.env.Environment;
import org.springframework.scheduling.annotation.EnableScheduling;

import io.github.pixee.security.SystemCommand;

import jakarta.annotation.PostConstruct;
import stirling.software.SPDF.config.ConfigInitializer;
import stirling.software.SPDF.utils.GeneralUtils;

@SpringBootApplication
@EnableScheduling
public class SPdfApplication {

    @Autowired private Environment env;

    private static String serverPortStatic;

    @Value("${server.port:8080}")
    public void setServerPortStatic(String port) {
        SPdfApplication.serverPortStatic = port;
    }

    @PostConstruct
    public void init() {
        // Check if the BROWSER_OPEN environment variable is set to true
        String browserOpenEnv = env.getProperty("BROWSER_OPEN");
        boolean browserOpen = browserOpenEnv != null && "true".equalsIgnoreCase(browserOpenEnv);

        if (browserOpen) {
            try {
                String url = "http://localhost:" + getNonStaticPort();

                String os = System.getProperty("os.name").toLowerCase();
                Runtime rt = Runtime.getRuntime();
                if (os.contains("win")) {
                    // For Windows
                    SystemCommand.runCommand(rt, "rundll32 url.dll,FileProtocolHandler " + url);
                }
            } catch (Exception e) {
                e.printStackTrace();
            }
        }
    }

    public static void main(String[] args) {
        SpringApplication app = new SpringApplication(SPdfApplication.class);
        app.addInitializers(new ConfigInitializer());
        if (Files.exists(Paths.get("configs/settings.yml"))) {
            app.setDefaultProperties(
                    Collections.singletonMap(
                            "spring.config.additional-location", "file:configs/settings.yml"));
        } else {
            System.out.println(
                    "External configuration file 'configs/settings.yml' does not exist. Using default configuration and environment configuration instead.");
        }
        app.run(args);

        try {
            Thread.sleep(1000);
        } catch (InterruptedException e) {
            // TODO Auto-generated catch block
            e.printStackTrace();
        }

        GeneralUtils.createDir("customFiles/static/");
        GeneralUtils.createDir("customFiles/templates/");

        System.out.println("Stirling-PDF Started.");

        String url = "http://localhost:" + getStaticPort();
        System.out.println("Navigate to " + url);
    }

    public static String getStaticPort() {
        return serverPortStatic;
    }

    public String getNonStaticPort() {
        return serverPortStatic;
    }
}

```

---
https://github.com/Stirling-Tools/Stirling-PDF/blob/625275124a9b6d4a18cebf2c52f60bb584b9eccd/src/main/java/stirling/software/SPDF/controller/api/pipeline/PipelineProcessor.java#L67

Change to `String port = SPdfApplication.getStaticPort();`

---
https://github.com/Stirling-Tools/Stirling-PDF/blob/625275124a9b6d4a18cebf2c52f60bb584b9eccd/src/main/java/stirling/software/SPDF/controller/api/pipeline/ApiDocService.java#L39

Change to `String port = SPdfApplication.getStaticPort();`

### PR DESCRIPTION
**Pull request by Ludy87**
# Description

Fix: Overriding Default Server Port

Do not merge this PR until PR #853 is merged.

Closes #834

## Checklist:

- [x] I have read the [Contribution Guidelines](https://github.com/Stirling-Tools/Stirling-PDF/blob/main/CONTRIBUTING.md)
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings

## Contributor License Agreement

By submitting this pull request, I acknowledge and agree that my contributions will be included in Stirling-PDF and that they can be relicensed in the future under the MPL 2.0 (Mozilla Public License Version 2.0) license.

(This does not change the general open-source nature of Stirling-PDF, simply moving from one license to another license)

### CODE REVIEW

**Comment by sbplat** [comment]
Has this been tested with docker (default port, custom port, etc.)?

**Comment by Ludy87** [comment]
No, not in docker, I wouldn't know how to do that. I only tested it locally with `gradlew bootRun.`

**Review by Frooodle — APPROVED** [review]

File: src/main/java/stirling/software/SPDF/SPdfApplication.java [hunk 1]
```diff
@@ -26,6 +27,14 @@ public class SPdfApplication {
     private static final Logger logger = LoggerFactory.getLogger(SPdfApplication.class);
 
     @Autowired private Environment env;
+    @Autowired ApplicationProperties applicationProperties;
```
Comment (Frooodle) [inline src/main/java/stirling/software/SPDF/SPdfApplication.java]: whats applicationProperties for?

**Review by Frooodle — CHANGES_REQUESTED** [review]

File: src/main/java/stirling/software/SPDF/SPdfApplication.java (hunk 1 above)
Comment (Ludy87) [reply src/main/java/stirling/software/SPDF/SPdfApplication.java]: outdated import, i removed

**Review by sbplat — APPROVED** [review]

**Review by Frooodle — APPROVED** [review]

**Pull request merged**